cmd/walletd/node.go

```diff
@@ -97,6 +99,35 @@ func setupUPNP(ctx context.Context, port uint16, log *zap.Logger) (string, error
 	return d.ExternalIP()
 }
 
+func loadCustomNetwork(fp string) (*consensus.Network, types.Block, error) {
+	f, err := os.Open(fp)
+	if err != nil {
+		return nil, types.Block{}, fmt.Errorf("failed to open network file: %w", err)
+	}
+	defer f.Close()
+
+	var network struct {
+		Network consensus.Network `json:"network" yaml:"network"`
+		Genesis types.Block       `json:"genesis" yaml:"genesis"`
+	}
+
+	switch filepath.Ext(fp) {
+	case ".yml", ".yaml":
+		dec := yaml.NewDecoder(f)
+		dec.KnownFields(true)
+		if err := dec.Decode(&network); err != nil {
+			return nil, types.Block{}, fmt.Errorf("failed to decode YAML network file: %w", err)
+		}
+	case ".json":
```

Comment:
I don't mind having both options but I'd consider only doing `json`. We don't have annotations for yml and if we allow multiple ways to configure it we should at least provide one valid example for each.